### PR TITLE
Update Agent to V1.24.0

### PR DESCRIPTION
**Pull request by suneyz**
### Summary
ECS Agent release v1.24.0

### Testing

New tests cover the changes: 

### Licensing

This contribution is under the terms of the Apache 2.0 License:

### CODE REVIEW

File: packaging/ubuntu-trusty/debian/changelog [hunk 1]
```diff
@@ -1,3 +1,9 @@
+amazon-ecs-init (1.22.0-2) trusty; urgency=medium
```
Comment (jhaynes) [inline packaging/ubuntu-trusty/debian/changelog]: This should be 1.24.0-1

File: packaging/amazon-linux-ami/ecs-init.spec [hunk 2]
```diff
@@ -255,6 +255,10 @@ if [ -e %{running_semaphore} ]; then
 fi
 %endif
 
+%changelog
+* Fri Jan 4 2019 Eric Sun <yuzhusun@amazon.com> - 1.24.0-1
+- Cache Agent version 1.24.0
+
 %changelog
```
Comment (jhaynes) [inline packaging/amazon-linux-ami/ecs-init.spec]: This `%changelog` line needs to be removed.

File: packaging/ubuntu-trusty/debian/changelog (hunk 1 above)
Comment (suneyz) [reply packaging/ubuntu-trusty/debian/changelog]: good catch!

File: packaging/amazon-linux-ami/ecs-init.spec (hunk 2 above)
Comment (suneyz) [reply packaging/amazon-linux-ami/ecs-init.spec]: Removed

**Comment by jhaynes** [comment]
approved once tests finish.

**Pull request merged**